### PR TITLE
Fix window position

### PR DESCRIPTION
**Pull request by sk7**
workaround for https://jira.ubergrape.com/browse/GRAPE-14546

### CODE REVIEW

**Comment by sk7** [comment]
commits are from IEUser because I did the changes on my Test-windows and committed it there

File: app/src/app/initApp.js [hunk 1]
```diff
@@ -72,19 +73,25 @@ export default () => {
       }
     )
 
+    // workaround to solve invisible app problem when windows was closed on
```
Comment (kof) [inline app/src/app/initApp.js]: you mean window was closed?

File: app/src/app/initApp.js [hunk 2]
```diff
@@ -72,19 +73,25 @@ export default () => {
...
+    // another monitor previously but Grape starts without the monitor now
+    if (state.dimensions.x < 0 || state.dimensions.x < 0) {
```
Comment (kof) [inline app/src/app/initApp.js]: I don't understand your comment and your code is broken.

File: app/src/app/initApp.js [hunk 3]
```diff
@@ -72,19 +73,25 @@ export default () => {
...
+      state.prefs = Object.assign(
+        state.prefs,
+        {x: 0, y: 0}
+      )
```
Comment (kof) [inline app/src/app/initApp.js]: You don't need  `state.prefs = `. I am assuming you wanted to set state.prefs.x and state.prefs.y. Also I would put it into some namespace, prefs.x is not clear what those coordinates are.

File: app/src/electron/windowStateKeeper.js [hunk 4]
```diff
@@ -1,48 +0,0 @@
-// Simple module to help you remember the size and position of windows.
-// Can be used for more than one window, just construct many
-// instances of it and give each different name.
-
-import { app } from 'electron';
-import jetpack from 'fs-jetpack';
-
-export default function (name, defaults) {
-
-    var userDataDir = jetpack.cwd(app.getPath('userData'));
```
Comment (kof) [inline app/src/electron/windowStateKeeper.js]: It has been saving in userData, where is it saving now?

File: app/src/app/initApp.js (hunk 2 above)
Comment (sk7) [reply app/src/app/initApp.js]: oh shit!

File: app/src/electron/windowStateKeeper.js (hunk 4 above)
Comment (sk7) [reply app/src/electron/windowStateKeeper.js]: this is handled by a new package I installed (electron-window-state), it's saving a file in the same location that has almost the same filename (window-state.json vs window-state-main.json previously)

File: app/src/app/initApp.js (hunk 3 above)
Comment (sk7) [reply app/src/app/initApp.js]: `new BrowserWindow(state.prefs)` needs `x` and `y` in the object supplied at frist argument. how would you do it?

File: app/src/app/initApp.js [hunk 5]
```diff
@@ -72,19 +73,27 @@ export default () => {
...
     )
 
+    // workaround to solve problem:
+    // 1. move grape window to second screen
+    // 2. quit grape -> x or y might be negative now in windw-state.json
+    // 3. unplug monitor
+    // 4. start Grape again
+    // see GRAPE-14546
+    if (state.dimensions.x < 0 || state.dimensions.y < 0) {
+      state.prefs.x = 0;
```
Comment (kof) [inline app/src/app/initApp.js]: are you using this values somewhere? Btw. we don't use semicolons.

**Comment by kof** [comment]
Please use npm run lint

**Comment by kof** [comment]
Actually you should have linted just your changes, not everything, because now if you run `npm run lint` instead of npm run lint:staged you will get a lot of errors.

File: app/src/app/initApp.js (hunk 5 above)
Comment (sk7) [reply app/src/app/initApp.js]: `new BrowserWindow(state.prefs)` expects those values

File: app/src/app/initApp.js [hunk 6]
```diff
@@ -72,19 +73,27 @@ export default () => {
...
+
     // set global to be accessible from webpage
     global.isNotificationSupported = isNotificationSupported()
     global.host = state.host
     global.grapeHost = env.host
     global.chooseDomainDisabled = env.chooseDomainDisabled
 
     state.mainWindow = new BrowserWindow(state.prefs)
-    if (state.dimensions.isMaximized && state.prefs.show) {
+    /* if (state.dimensions.isMaximized && state.prefs.show) {
```
Comment (kof) [inline app/src/app/initApp.js]: why is this commented out?Can it  be removed?

**Comment by kof** [comment]
you can reset the commits, run lint:staged and commit again

File: app/src/app/initApp.js (hunk 5 above)
Comment (sk7) [reply app/src/app/initApp.js]: removed semicolons

File: tasks/release/windows.js [hunk 7]
```diff
@@ -385,8 +385,8 @@ module.exports = function() {
         .then(packageBuiltApp)
         .then(finalize)
         .then(renameApp)
-        .then(createInstaller)
-        .then(cleanClutter)
-        .then(signApp)
+        //.then(createInstaller)
+        //.then(cleanClutter)
+//        .then(signApp)
```
Comment (sk7) [inline tasks/release/windows.js]: damnit

File: app/src/app/initApp.js (hunk 5 above)
Comment (kof) [reply app/src/app/initApp.js]: ok

File: app/src/app/initApp.js (hunk 6 above)
Comment (sk7) [reply app/src/app/initApp.js]: will reactivate, turns out this is needeed to handle maximized windows. I had the wrong impression that electron-window-state handles that automatically

**Pull request merged**